server-spi/src/main/java/org/keycloak/userprofile/UserProfileDecorator.java

```diff
@@ -32,5 +34,5 @@ public interface UserProfileDecorator {
      *
      * @param metadata to decorate
      */
-    void decorateUserProfile(RealmModel realm, UserProfileMetadata metadata);
+    List<AttributeMetadata> decorateUserProfile(String providerId, RealmModel realm, UserProfileMetadata metadata);
```

Comment:
Do you really want that? I mean, we just introduced this interface and I'm pretty sure nobody is actually using it ..

Don't know ... But it is a good point.

Reply:
I don't want that. I want to not care about backwards compatibility :-) But I afraid we need to care about backwards compatibility even if I don't want that :-(

AFAIK the policy is, that any changes in `server-spi` in method signatures are problematic. Here for this case, the "advantage" is, that we don't yet document UserProfileDecorator (see https://github.com/keycloak/keycloak/issues/28237) . Sop hopefully not much people use that. But not 100% sure we can rely on that...

If not possible to avoid keeping existing signature method as deprecated, is it possible to at least document this in migration guide?

Reply:
Added documentation to the guide.